submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt

```diff
@@ -28,4 +33,11 @@ class SubmissionRepository(
     fun findProjectsByAccessTags(tags: List<AccessTag>) =
         submissionRepository.findByTypeAndAccNo(Project.value, tags.map { it.name })
             .map { submissionDbMapper.toSubmission(it) }
+
+    fun getSubmissionsByUser(userId: Long, filter: SubmissionFilter): List<Submission> {
+        val filterSpecs = SubmissionFilterSpecification(userId, filter)
+        return submissionRepository.findAll(filterSpecs.specification,
+            OffsetLimitPageable(filter.offset, filter.limit, Sort.by("releaseTime").descending())
```

Comment:
No need a custom implementation of `Pageable` we can use spring page request with calculation for page number based on offset `PageRequest.of(filter.offset / filter.limit, filter.limit)`